app/src/main/java/com/esri/arcgismaps/kotlin/sampleviewer/SampleViewerLauncherActivity.kt

```diff
@@ -0,0 +1,86 @@
+/* Copyright 2024 Esri
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.esri.arcgismaps.kotlin.sampleviewer
+
+import android.content.Intent
+import android.os.Bundle
+import android.os.Handler
+import android.os.Looper
+import android.util.Log
+import androidx.activity.ComponentActivity
+
+class SampleViewerLauncherActivity : ComponentActivity(), ExceptionListener {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setupExceptionHandler()
+        startMainActivity(null)
+    }
+
+    /**
+     *  Prepares extras for the intent with error information if a [throwable] is provided.
+     *  It then starts the [MainActivity] by creating an intent with necessary flag and extras.
+     */
+    private fun startMainActivity(throwable: Throwable?) {
+        val extras = Bundle()
+        if (throwable != null) {
+            extras.putStringArray(
+                /* key = */     "SampleViewerException",
+                /* value = */   arrayOf(throwable.message.toString(), throwable.cause.toString())
```

Comment:
Do we want to separate out additionalMessage or any others? Might aid in formatting with /n's between each in the description section of the MessageDialog.

Maybe just fine as is...

Reply:
The `Throwable` does not give us properties to access the `additionalMessage` individually but rather is concentrated into the `.cause` property. If the override param was an `Exception` it could have been possible to cast to an `ArcGISException`  to make this work.

Reply:
Leaving it as is 👍